### PR TITLE
rpm: update to 4.18.1.

### PR DESCRIPTION
**Pull request by ismaell**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-musl)

### CODE REVIEW

File: srcpkgs/rpm/template [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Template file for 'rpm'
 pkgname=rpm
-version=4.17.1
-revision=3
+version=4.18.0
```
Comment (classabbyamp) [inline srcpkgs/rpm/template:3]: .1 is available

**Review by classabbyamp — COMMENTED** [review]
add [this patch](https://github.com/rpm-software-management/rpm/commit/d8eee23e1f2d944a9bcef5044d43808ba21bb83a.patch) and remove musl-legacy-compat from the template

**Comment by ismaell** [comment]
> add [this patch](https://github.com/rpm-software-management/rpm/commit/d8eee23e1f2d944a9bcef5044d43808ba21bb83a.patch) and remove musl-legacy-compat from the template

I was planning on submitting that separately, but I guess I could just include it here

**Comment by classabbyamp** [comment]
> note that the patch was already present in the 4.18.0 release.

weird, it didn't show up as in the tag in github's UI

**Comment by classabbyamp** [comment]
ah ok

overall it's fine to do both an update and some packaging fix at once

**Pull request merged**